tests/o11y/o11y.go

```diff
@@ -0,0 +1,63 @@
+package o11y
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
+)
+
+const (
+	O11yUser = "o11y-e2e"
+)
+
+var _ = framework.O11ySuiteDescribe("OllY E2E tests", Label("o11y", "HACBS"), Pending, func() {
+
+	defer GinkgoRecover()
+	var f *framework.Framework
+	var err error
+
+	Describe("O11y test", Pending, func() {
+		var testNamespace string
+
+		BeforeAll(func() {
+
+			f, err = framework.NewFramework(O11yUser)
+			Expect(err).NotTo(HaveOccurred())
+			testNamespace = f.UserNamespace
+
+			_, err := f.AsKubeAdmin.CommonController.CreateTestNamespace(testNamespace)
+			Expect(err).NotTo(HaveOccurred(), "Error when creating/updating '%s' namespace: %v", testNamespace, err)
+
+		})
+		AfterAll(func() {
+			if !CurrentSpecReport().Failed() {
+				err := f.AsKubeAdmin.O11yController.DeleteAllPipelineRunsAndTasks(testNamespace)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(f.SandboxController.DeleteUserSignup(f.UserName)).NotTo(BeFalse())
+			}
+		})
+
+		It("E2E test to get podname and size from prometheus query", Pending, func() {
+			podNameRegex := ".*-upload-50mb-pod"
+
+			query := fmt.Sprintf("last_over_time(container_network_transmit_bytes_total{namespace='%s', pod=~'%s'}[1h])", testNamespace, podNameRegex)
+			result, err := f.AsKubeAdmin.O11yController.GetMetrics(query)
+			Expect(err).NotTo(HaveOccurred())
+
+			podNames, err := f.AsKubeAdmin.O11yController.GetRegexPodNames(podNameRegex, result)
+			Expect(err).NotTo(HaveOccurred())
+			GinkgoWriter.Printf("Pods: %s /n", podNames)
+
+			podNamesWithSize, err := f.AsKubeAdmin.O11yController.GetPodNamewithSize(podNameRegex, result)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
What's the purpose of filtering by the pod name regex in both the PromQL query and multiple times using `GetRegexPodNames` and `GetPodNamewithSize` here? IMO, it would be cleaner to leave the regex in the query and instead assert that it returns an expected result size. There could also be an assertion that ensures all results match the same pod. Then I don't see the need for `GetRegexPodNames` and `GetPodNamewithSize` and you could loop through `results` below this.